src/main/java/uk/gov/companieshouse/confirmationstatementapi/eligibility/impl/CompanyTypeValidationForWebFiling.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.companieshouse.confirmationstatementapi.eligibility.impl;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityFailureReason;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+
+import java.util.Set;
+
+public class CompanyTypeValidationForWebFiling implements EligibilityRule<CompanyProfileApi> {
+
+    private final Set<String> allowedTypes;
+
+    public CompanyTypeValidationForWebFiling(Set<String> allowedTypes) {
+        this.allowedTypes = allowedTypes;
+    }
+
+    @Override
+    public void validate(CompanyProfileApi profileToValidate) throws EligibilityException {
+        var status = profileToValidate.getType();
+
+        if (!allowedTypes.contains(status)) {
```

Comment:
similarly to a comment on Alfred's pr: https://github.com/companieshouse/confirmation-statement-api/pull/21#discussion_r632430717
It may make more sense to "disallow webfiling" companies.
Instead of Allowing not webfiling companies, if that makes sense?

So create a set of types that are webfiling only.
And do if webfilingTypes.contains(profileToValidate.getType()) -> Throw Ex.

Also, may want to rename var on line 20 from status to type